File: JAVA/PokerProtocol/src/bluffinmuffin/protocol/commands2/lobby/training/CreateTrainingTableResponse.java
package bluffinmuffin.protocol.commands2.lobby.training;

import java.util.StringTokenizer;

import bluffinmuffin.protocol.commands2.lobby.AbstractLobbyResponse;

public class CreateTrainingTableResponse extends AbstractLobbyResponse<CreateTrainingTableCommand>
{
    @Override
    protected String getCommandName()
    {
        return CreateTrainingTableResponse.COMMAND_NAME;
    }
    
    public static String COMMAND_NAME = "lobbyTRAINING_CREATE_TABLE_RESPONSE";
    
    private final int m_ResponsePort;
    
    public CreateTrainingTableResponse(StringTokenizer argsToken)
    {
        super(new CreateTrainingTableCommand(argsToken));
        m_ResponsePort = Integer.parseInt(argsToken.nextToken());
    }
    
    public CreateTrainingTableResponse(CreateTrainingTableCommand command, int responsePort)
    {
        super(command);
        m_ResponsePort = responsePort;
    }
    
    @Override
    public void encode(StringBuilder sb)
    {
        super.encode(sb);
        append(sb, m_ResponsePort);
    }
    
    public int getResponsePort()
    {
        return m_ResponsePort;
    }
    
}

File: trunk/C#/PokerProtocol/Commands/Lobby/Training/CreateTrainingTableResponse.cs
using System;
using System.Collections.Generic;
using System.Text;
using EricUtility;
using EricUtility.Networking.Commands;
using Newtonsoft.Json.Linq;

namespace PokerProtocol.Commands.Lobby.Training
{
    public class CreateTrainingTableResponse : AbstractLobbyResponse<CreateTrainingTableCommand>
    {
        public static string COMMAND_NAME = "lobbyTRAINING_CREATE_TABLE_RESPONSE";

        public int Port { get; set; }

        public CreateTrainingTableResponse()
            : base()
        {
        }

        public CreateTrainingTableResponse(CreateTrainingTableCommand command, int port)
            : base(command)
        {
            Port = port;
        }
    }
}
